provider/job_utils.py

```diff
@@ -244,11 +244,14 @@ def is_block_job_running(vm, jobid, tmo=200):
         if not job:
             LOG_JOB.debug('job %s cancelled unexpectedly', jobid)
             break
-        elif job['status'] != "running":
+        elif job['status'] not in ["running", "pending", "ready"]:
```

Comment:
I am afraid here the job status should only be 'running', we are using the function dedicated to a running job, e.g. 
1. check job is running
2. do some operations
3. check job is still running
The job's status after 2 may be ready or pending, so you add pending/ready check here, but consider 1, it still returns true if the status is ready/pending, but that's not expected.
We can set the job speed to a small one to avoid a running job enters into pending/ready state.

Reply:
1. When we set block-job speed to 0, then we check job is running, the first round check, we get "running"+"offset=None" we set the offset=offset1, then we continue, sometimes, for the second round check, job reach "pending/ready status" with offset > offset1(depends on test machine's performance), then the case should pass.
2. Or sometimes, we set a high speed, then we reset the speed to 0, for the first round check, the job may in "pending/ready" status, so offset checking is meaningless, we need to skip it, and also for this scenario, the case should pass.